torch/csrc/distributed/c10d/TCPStore.cpp

```diff
@@ -409,15 +409,29 @@ void TCPStore::_splitSet(
   buffer.flush();
 }
 
-void TCPStore::set(const std::string& key, const std::vector<uint8_t>& data) {
+void TCPStore::doSet(
+    const std::string& key,
+    const std::vector<uint8_t>& data,
+    bool usePrefix) {
   detail::timing_guard tguard(clientCounters_["set"]);
   const std::lock_guard<std::mutex> lock(activeOpLock_);
   detail::SendBuffer buffer(*client_, detail::QueryType::SET);
-  buffer.appendString(keyPrefix_ + key);
+  auto keyToUse = usePrefix ? keyPrefix_ + key : key;
+  buffer.appendString(keyToUse);
   buffer.appendBytes(data);
   buffer.flush();
 }
 
+void TCPStore::set(const std::string& key, const std::vector<uint8_t>& data) {
```

Comment:
did you consider changing the API for set/check to expose a new kwarg (global=False) instead of adding 2 new APIs?

Reply:
That is a good point... Although the flag basically just negate what PrefixStore is doing, it does not make a big difference between new API I proposed here..